src/core/lib/event_engine/poller.h

```diff
@@ -30,20 +30,23 @@ namespace experimental {
 // Work(...).
 class Poller {
  public:
-  // This initial vector size may need to be tuned
-  using Events = absl::InlinedVector<EventEngine::Closure*, 5>;
+  struct Ok {};
   struct DeadlineExceeded {};
   struct Kicked {};
-  using WorkResult = absl::variant<Events, DeadlineExceeded, Kicked>;
+  using WorkResult = absl::variant<Ok, DeadlineExceeded, Kicked>;
 
   virtual ~Poller() = default;
-  // Poll once for events, returning a collection of Closures to be executed.
+  // Poll once for events and process received events. The callback function
+  // is expected to be invoked prior to processing received events.
```

Comment:
Please be explicit about what the callback's responsibility is, which is to schedule another thread that calls Work. And because of that, I think it would be better advice to just say "The implementation should run the next polling cycle as quickly as possible to ensure continuous polling." And that may not mean calling it immediately!

For example, say this self-scheduling callback uses the same executor that's used inside the Work implementation to schedule read/write callbacks. If it's a LIFO executor, calling that function first may mean that the polling Work thread runs *after* read/write callbacks are processed. I'd let the implementation decide the best course of events, and otherwise make the API contract clear about whether it's a blocking or non-blocking call. If it's a blocking call, then the Work implementation has more control of the actual ordering, so I'd recommend that.

Reply:
Done.